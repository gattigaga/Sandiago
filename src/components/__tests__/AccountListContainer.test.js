import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import toJSON from "enzyme-to-json";

import ConnectedComponent, {
  AccountListContainer
} from "../AccountListContainer";

describe("AccountListContainer", () => {
  Date.now = jest.fn(() => 1529122840652);

  const mockStore = configureStore();

  const setup = (props = {}, isConnected = true) => {
    const store = mockStore();
    store.dispatch = jest.fn();

    const wrapper = isConnected
      ? shallow(<ConnectedComponent store={store} />)
      : shallow(<AccountListContainer {...props} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders with accounts", () => {
    const items = [
      {
        id: "1",
        name: "Facebook",
        description: "This is my first facebook account",
        date: Date.now()
      },
      {
        id: "2",
        name: "Twitter",
        description: "This is my first twitter account",
        date: Date.now()
      },
      {
        id: "3",
        name: "Google",
        description: "This is my first google account",
        date: Date.now()
      }
    ];
    const { wrapper } = setup({ items }, false);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders without accounts", () => {
    const { wrapper } = setup({}, false);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should maps state and dispatch to props", () => {
    const { wrapper } = setup();

    expect(wrapper.props()).toMatchObject({
      items: []
    });
  });
});
