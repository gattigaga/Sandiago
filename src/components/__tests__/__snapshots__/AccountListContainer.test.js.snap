// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountListContainer should renders with accounts 1`] = `
<Styled(Component)>
  <Styled(ScrollView)>
    <AccountList
      items={
        Array [
          Object {
            "date": 1529122840652,
            "description": "This is my first facebook account",
            "id": "1",
            "name": "Facebook",
          },
          Object {
            "date": 1529122840652,
            "description": "This is my first twitter account",
            "id": "2",
            "name": "Twitter",
          },
          Object {
            "date": 1529122840652,
            "description": "This is my first google account",
            "id": "3",
            "name": "Google",
          },
        ]
      }
      onPress={[Function]}
    />
  </Styled(ScrollView)>
</Styled(Component)>
`;

exports[`AccountListContainer should renders without accounts 1`] = `
<Styled(Component)>
  <Styled(Text)>
    No Accounts were found.
  </Styled(Text)>
</Styled(Component)>
`;
